governance/program/src/state/governance.rs

```diff
@@ -104,7 +125,13 @@ pub struct GovernanceV2 {
     /// Reserved space for versions v2 and onwards
     /// Note 1: V1 accounts must be resized before using this space
     /// Note 2: The reserved space should be used from the end to also allow the config to grow if needed
-    pub reserved_v2: Reserved120,
+    pub reserved_v2: Reserved115,
+
+    /// The number of required signatories for proposals in the Governance
+    pub signatories_count: u8,
+
+    /// Required to track whether required signatories have changed since a proposal's creation
+    pub signatories_nonce: u32,
```

Comment:
This is a very good idea but we should not just scope it to a single setting of required signatories. Any change to governance config should invalidate all inflight proposals. Let's move it out of the scope of this PR and do as a separate one please.